types/events.go

```diff
@@ -140,112 +139,54 @@ type EventDataVote struct {
 	Vote *Vote
 }
 
-func (_ EventDataNewBlock) AssertIsTMEventData()       {}
-func (_ EventDataNewBlockHeader) AssertIsTMEventData() {}
-func (_ EventDataTx) AssertIsTMEventData()             {}
-func (_ EventDataRoundState) AssertIsTMEventData()     {}
-func (_ EventDataVote) AssertIsTMEventData()           {}
-
+func (_ EventDataNewBlock) AssertIsTMEventData()          {}
```

Comment:
We don't need these anymore, and perhaps if we still do, let's use type assertions.